pallets/metamask/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    crypto::ecdsa::ECDSAExt, dispatch::GetDispatchInfo, traits::UnfilteredDispatchable,
+};
+use sp_core::{
+    ecdsa::{Public, Signature},
+    H160, H256,
+};
+use sp_io::crypto::secp256k1_ecdsa_recover_compressed;
+use sp_std::{boxed::Box, vec::Vec};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+pub use weights::WeightInfo;
+
+mod eip712;
+use eip712::{Domain, Payload, TypedData};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::{pallet_prelude::*, traits::Get};
+    use frame_system::{pallet_prelude::*, Account, RawOrigin};
+
+    #[pallet::config(with_default)]
+    pub trait Config: frame_system::Config + pallet_evm_chain_id::Config {
+        type Sender: Parameter + Into<sp_core::H160> + Into<Self::AccountId>;
```

Comment:
I see you depend on `Into<sp_core::H160>` and `Into<sp_core::U256>` here, but is there absolutely no way to use `frame_system::Config::AccountId` and `frame_system::Config::Nonce` here?

Reply:
Both `frame_system::Config::AccountId` and `frame_system::Config::Nonce` types have no any useful type constraint to convert them into required specific types.
Looks like there are another way to make constraints of use this pallet from runtime with specific types range. I mean using `where <T as Config>::AccountId: SomeConstraint` clause. But IMO this approach is hard to use due to macro-oriented nature of pallets and obscure error messages.

Here the both types signatures:
```rust
type AccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeDisplay + Ord + MaxEncodedLen;
type Nonce: Parameter + Member + MaybeSerializeDeserialize + Debug + Default + MaybeDisplay + AtLeast32Bit + Copy + MaxEncodedLen;
```

Reply:
And yeah, these types are aliased in runtime as the same types used for `frame_system::Config::AccountId` and `frame_system::Config::Nonce`.
I tried using `IsType<AccountId>` trait to specify the exact type but that doesn't work either, or I'm doing it wrong.

Reply:
Ok then